### PR TITLE
Closes: #192 Use the specific config structure for each command

### LINKED ISSUE #192: Use the specific config structure for each command
Now there are different flagsets for different commands, but for each command, it uses a global config structure:

```
type Config struct {
    Version bool
    Agent     Agent     `skip:“true”`
    Server    Server    `skip:“true”`
    Convert   Convert   `skip:“true”`
    Exec      Exec      `skip:“true”`
    DbManager DbManager `skip:“true”`
}
```

just like the server command, the function is like this:
```
func startServer(cfg *config.Config)
```

i think it's better to use its own config for each command, the function should be like this:
```
func startServer(cfg *config.Server)
```

### PR DESCRIPTION
**Pull request by AdrK**
#192

### CODE REVIEW

File: pkg/agent/cli/agent.go [hunk 1]
```diff
@@ -15,19 +15,19 @@ import (
...
-	cfg            *config.Config
+	cfg            *config.Agent
 	cs             *csock.CSock
 	activeProfiles map[int]*agent.ProfileSession
 	id             id.ID
 	u              upstream.Upstream
 }
 
-func New(cfg *config.Config) *Agent {
+func New(cfg *config.Agent) *Agent {
```
Comment (AdrK) [inline pkg/agent/cli/agent.go:25]: @petethepig Seems that this is unused, is that possible?

File: pkg/agent/cli/agent.go [hunk 2]
```diff
@@ -15,19 +15,19 @@ import (
 )
 
 type Agent struct {
-	cfg            *config.Config
```
Comment (AdrK) [inline pkg/agent/cli/agent.go:18]: @petethepig 
Do we want to keep the config.Config struct?
If my understanding is correct, the lifetime of all the config structs are withing the `Start(cfg *config.Config)` function in cli.go, so maybe we don't need an extra container for all the configs?

File: pkg/agent/cli/agent.go (hunk 1 above)
Comment (petethepig) [reply pkg/agent/cli/agent.go:26]: Yeah, I think we'll delete it soon

File: pkg/agent/cli/agent.go (hunk 2 above)
Comment (petethepig) [reply pkg/agent/cli/agent.go:19]: Yeah, good point, we might do that in the future

**Pull request merged**